### ISSUE TITLE
Wrong error message for Task.Run with invalid argument

### ISSUE THREAD

**Issue by svick** [dotnet/roslyn#10672]
**Version Used**: VS 2015 U2

**Steps to Reproduce**:

Write the following erroneous C# statement:

```c#
Task.Run(() => { }, TaskCreationOptions.LongRunning);
```

**Expected Behavior**:

A reasonable error, probably:

> CS1503 Argument 2: cannot convert from 'System.Threading.Tasks.TaskCreationOptions' to 'System.Threading.CancellationToken'

**Actual Behavior**:

Incorrect error message:

> CS1643: Not all code paths return a value in lambda expression of type 'Func\<Task>'

This is possibly related to #8603.

**Comment by gafter** [comment]
@jaredpar Who is currently on this issue?
